pkg/controller/changetierrequest/changetierrequest_controller_test.go

```diff
@@ -274,15 +273,17 @@ func NewNSTemplateTier(tierName, revision, clusterResourcesRevision string, nsTy
 	namespaces := make([]v1alpha1.NSTemplateTierNamespace, len(nsTypes))
 	for i, nsType := range nsTypes {
 		namespaces[i] = v1alpha1.NSTemplateTierNamespace{
-			Type:     nsType,
-			Revision: revision,
-			Template: templatev1.Template{},
+			Type:        nsType,
+			Revision:    revision,
+			Template:    templatev1.Template{},
+			TemplateRef: nstemplatetiers.NewTierTemplateName(tierName, nsType, revision),
 		}
 	}
 	var clusterResources *v1alpha1.NSTemplateTierClusterResources
 	if clusterResourcesRevision != "" {
 		clusterResources = &v1alpha1.NSTemplateTierClusterResources{
-			Revision: clusterResourcesRevision,
+			Revision:    clusterResourcesRevision,
+			TemplateRef: nstemplatetiers.NewTierTemplateName(tierName, nstemplatetiers.ClusterResources, revision),
```

Comment:
you should use clusterResourcesRevision here
```suggestion
			TemplateRef: nstemplatetiers.NewTierTemplateName(tierName, nstemplatetiers.ClusterResources, clusterResourcesRevision),
```